File: v2/src/main/java/com/api/v2/customers/services/CustomerRetrievalService.java
package com.api.v2.customers.services;

import com.api.v2.common.Id;
import com.api.v2.customers.dtos.CustomerResponseDto;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface CustomerRetrievalService {
    Mono<CustomerResponseDto> findById(@Id String id);
    Flux<CustomerResponseDto> findAll();
}

File: v2/src/main/kotlin/com/api/v2/customers/services/CustomerRetrievalService.kt
package com.api.v2.customers.services

import com.api.v2.customers.responses.CustomerResponseDTO
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.http.ResponseEntity

interface CustomerRetrievalService {
    fun findById(id: String): ResponseEntity<CustomerResponseDTO>
    fun findAll(pageable: Pageable): ResponseEntity<Page<CustomerResponseDTO>>
}